### PR TITLE
Merge gd-merge:release-to-develop into develop

### PR DESCRIPTION
**Pull request by gd-merge**

### CODE REVIEW

**Comment by gdgate** [comment]
Build in pipeline check aborted.

File: lib/gooddata/lcm/helpers/check_helper.rb [hunk 1]
```diff
@@ -18,20 +18,32 @@ def check_params(specification, params)
               if specification[param_name][:opts][:default]
                 params[param_name] = specification[param_name][:opts][:default]
               elsif specification[param_name][:opts][:required]
-                fail("Mandatory parameter '#{param_name}' of type '#{type}' is not specified")
+                if ENV['RSPEC_ENV'] == 'test'
```
Comment (yenkins) [inline lib/gooddata/lcm/helpers/check_helper.rb]: Calls 'env['rspec_env'] == 'test'' 3 times (DuplicateMethodCall)

Comment (yenkins) [inline lib/gooddata/lcm/helpers/check_helper.rb]: Calls 'env['rspec_env']' 3 times (DuplicateMethodCall)

Comment (yenkins) [inline lib/gooddata/lcm/helpers/check_helper.rb]: Tests 'env['rspec_env'] == 'test'' at least 3 times (RepeatedConditional)

File: lib/gooddata/lcm/helpers/check_helper.rb [hunk 2]
```diff
@@ -18,20 +18,32 @@ def check_params(specification, params)
...
+                  fail "Mandatory parameter '#{param_name}' of type '#{type}' is not specified"
+                else
+                  GoodData.logger.error("Mandatory parameter '#{param_name}' of type '#{type}' is not specified")
```
Comment (yenkins) [inline lib/gooddata/lcm/helpers/check_helper.rb]: Calls 'gooddata.logger' 4 times (DuplicateMethodCall)

File: lib/gooddata/lcm/helpers/check_helper.rb [hunk 3]
```diff
@@ -18,20 +18,32 @@ def check_params(specification, params)
...
+                if ENV['RSPEC_ENV'] == 'test'
+                  fail "Expected parameter '#{param_name}' to be kind of '#{type}', got '#{value.class.name}'"
+                else
+                  GoodData.logger.error("Expected parameter '#{param_name}' to be kind of '#{type}', got '#{value.class.name}'")
+                end
               end
 
               if specification[param_name][:opts][:deprecated]
-                puts "WARNING: Parameter '#{param_name}' is deprecated. Please use '#{specification[param_name][:opts][:replacement]}' instead."
+                GoodData.logger.warn("WARNING: Parameter '#{param_name}' is deprecated. Please use '#{specification[param_name][:opts][:replacement]}' instead.")
```
Comment (yenkins) [inline lib/gooddata/lcm/helpers/check_helper.rb]: Calls 'specification[param_name]' 6 times (DuplicateMethodCall)

Comment (yenkins) [inline lib/gooddata/lcm/helpers/check_helper.rb]: Calls 'specification[param_name][:opts]' 5 times (DuplicateMethodCall)

File: lib/gooddata/lcm/helpers/check_helper.rb [hunk 4]
```diff
@@ -18,20 +18,32 @@ def check_params(specification, params)
...
+                else
+                  GoodData.logger.error("Mandatory parameter '#{param_name}' of type '#{type}' is not specified")
+                end
               end
             else
               if type.class.const_get(:CATEGORY) == :complex && !value.is_a?(Hash)
                 puts JSON.pretty_generate(params)
-                fail "Expected parameter '#{param_name}' to be kind of '#{type}', got '#{value.class.name}'"
+                if ENV['RSPEC_ENV'] == 'test'
+                  fail "Expected parameter '#{param_name}' to be kind of '#{type}', got '#{value.class.name}'"
```
Comment (yenkins) [inline lib/gooddata/lcm/helpers/check_helper.rb]: Calls 'value.class' 4 times (DuplicateMethodCall)

Comment (yenkins) [inline lib/gooddata/lcm/helpers/check_helper.rb]: Calls 'value.class.name' 2 times (DuplicateMethodCall)

Comment (yenkins) [inline lib/gooddata/lcm/helpers/check_helper.rb]: Refers to 'value' more than self (maybe move it to another class?) (FeatureEnvy)

File: lib/gooddata/lcm/types/scalar/integer.rb [hunk 5]
```diff
@@ -13,8 +13,8 @@ class IntegerType < BaseType
         CATEGORY = :scalar
         INTERNAL_TYPE = Integer
 
-        def check(_value)
-          true
+        def check(value)
```
Comment (yenkins) [inline lib/gooddata/lcm/types/scalar/integer.rb]: Doesn't depend on instance state (maybe move it to another class?) (UtilityFunction)

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-merge-compliance-check/21/ : SUCCESS in 13s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/895/ : SUCCESS in 51s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/1163/ : SUCCESS in 51s
- https://ci.intgdc.com/job/gooddata-ruby-integration-tests-zuul/1355/ : SUCCESS in 55m 52s
- https://ci.intgdc.com/job/gooddata-ruby-lcm-integration-tests-zuul/1118/ : SUCCESS in 34m 40s

**Pull request merged**